This patch is based on <https://github.com/lathiat/avahi/pull/281>.

build-db is present in the upstream git repo, but not in the source tarball,
and the build fails if it is not present.

diff -aruN old/service-type-database/build-db new/service-type-database/build-db
--- old/service-type-database/build-db	1970-01-01 01:00:00.000000000 +0100
+++ new/service-type-database/build-db	2022-07-14 23:49:23.812061745 +0100
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+# -*-python-*-
+# This file is part of avahi.
+#
+# avahi is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# avahi is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+# or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with avahi; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+# USA.
+
+try:
+    import anydbm as dbm
+except ImportError:
+    import dbm
+
+import sys
+
+if len(sys.argv) > 1:
+    infn = sys.argv[1]
+else:
+    infn = "service-types"
+
+if len(sys.argv) > 2:
+    outfn = sys.argv[2]
+else:
+    outfn = infn + ".db"
+
+db = dbm.open(outfn, "n")
+
+for ln in open(infn, "r"):
+    ln = ln.strip(" \r\n\t")
+    
+    if ln == "" or ln.startswith("#"):
+        continue
+
+    t, n = ln.split(":", 1)
+
+    db[t.strip()] = n.strip()
+
+db.close()
